Applications/Spire/Source/Ui/ComboBox.cpp

```diff
@@ -238,6 +241,13 @@ void ComboBox::submit(const QString& query, bool is_passive) {
   m_submit_signal(value);
 }
 
+void ComboBox::on_current(const std::any& current) {
+  auto text = displayTextAny(current);
+  if(text != m_input_box->get_current()->get()) {
```

Comment:
Do not use text to compare for equality, a value can have multiple text representations, for example "5", "5.0", "5.000" all represent the same value but have different text representations. You will need to test for true equality.